tests/unit/s2n_certificate_test.c

```diff
@@ -92,6 +139,137 @@ int main(int argc, char **argv)
         EXPECT_BYTEARRAY_EQUAL(out_cert_der, cert->raw.data, cert_len);
     }
 
+    /* Test s2n_connection_get_peer_cert_chain */
+    {
+        /* Disable TLS 1.3 to work around cert validation setup as it is not required for the following tests. 
+         * Note that for TLS1.3 Certificate extensions is a field https://tools.ietf.org/html/rfc8446#section-4.4.2 
+         * and requires additional setup. */
+        EXPECT_SUCCESS(s2n_disable_tls13());
```

Comment:
All this setup can probably be replaced with just doing a valid handshake, with and without client auth, before calling the new API. That would also make me confident that the test setup is actually correct / matches what will happen in practice.

Reply:
Yes, I was trying to keep it to a unit test, and as we discussed - a functional one may be better suited for clarity.

Reply:
`s2n_connection_get_peer_cert_chain` API requires x509 verification which is a part of client auth and has to be set. Without client auth `s2n_connection_get_peer_cert_chain` cannot obtain the validated certificate.